engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoObject.java

```diff
@@ -0,0 +1,6 @@
+package org.enso.interpreter.runtime.data;
+
+import com.oracle.truffle.api.interop.TruffleObject;
+
+/** All non-primitive Enso types extends from {@code EnsoObject}. */
+public interface EnsoObject extends TruffleObject {}
```

Comment:
Having such a marker interface feels like a good idea. Could you look for the Enso builtin type classes that currently implement `TruffleObject` and refactor them to implement `EnsoObject` instead?

Reply:
Done in 91204f497ecb73fcd865a328403b855833505bf9